The North Star Expeditionary Vessel - A Second Wind

Cheshify opened the pull request:
## About The Pull Request
A new map for TGstation, in the works! It has 4 fucking Z levels, a massive expansive maintenance with unique designs, and some unique code features in the works.

To Do:
- [x] Update the Map to Modern TG
- [x] Local Tests
- [x] Work on Map Optimizations
- [x] Run Live Tests

Fikou has greatly helped with creating an important flavour aspect of this map, Trek Uniforms on anyone who joins! See the forum thread for more. This includes the framework for innate station traits, station traits loaded as long as it's in a map's json

Here's the forum dev thread there are screenshots there.
https://tgstation13.org/phpBB/viewtopic.php?p=657252#p657252

### Mapping March
Ckey to receive rewards: Cheshify

## Why It's Good For The Game
So, this is the North Star. An effort taking multiple mappers and of 9~ months of hard work. This map was not initially designed for TGstation, but always designed for TGstation code. The process of retooling the map for TGstation was an absolute joy and I feel like the map definitely has it's niche as a massive and unique experience for it's players.

I adore this map, it's gorgeous, has a unique aesthetic, and a number of very funny interactions with multi-Z. The PR comes packed with unique mechanics for future mappers (innate station traits!), a number of map-fitting shuttles, and a fun spacefaring uniform gimmick for the crew.

**This is my second attempt at bringing this map into rotation. It was initially closed due to concerns about maptick and performance, as I wasn't willing to push for a map to be added to the repository if it didn't function to my own standards. I've been informed by a number of coders far better than I that optimizations are arriving and enroute, so I think it's time to dust her off and set sail for another journey.**

**Quick Disclaimer: Due to some design decisions disagreed upon by the headcoder team and myself, the map will not be featuring unique roundstart uniforms, and despite my design intentions, the innate station trait features will be shelved for now.**

## Changelog
:cl: Cheshify, Fikou, Blue-Berry, Zytolg, InfiniteGalaxies, Striders, Sylphet, Riggle, Soal, Andry, Crit, Deranging, and Pumpkin0.
add: Nanotrasen's Newest Exploratory Vessel is now available! Meet the North Star!
add: More landmines, and a landmine random spawner.
add: energy barriers now have a regenerative subtype, fit for permanent installations.
code: Raised the number of possible level render to 4, check your preferences if needed to be reduced.
/:cl:

Cheshify committed:
Northstar Efforts 1

Cheshify committed:
Another update - plenty more to do

Cheshify committed:
More Ordnance work

Cheshify committed:
More-work

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
BIG edits, new areas, more content!

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
Playtest Ready!

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
sensor fix

Cheshify committed:
Last second medical tweaks

Cheshify committed:
mothblock suggestions, more airlocks, fixed CI

Cheshify committed:
2 not -2

Cheshify committed:
multiz_perf but working

Cheshify committed:
Correct at last

Cheshify committed:
multiz-done more Ci fixing

Cheshify committed:
Another Commit - Lotsa fixes

Cheshify committed:
Removes hidden space tile

Cheshify committed:
Communite Changes + Possible CI Fix

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
Dogs Updated

Cheshify committed:
innate traits (will not work)

Fikou committed:
whatever

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Fikou committed:
probably fix

Fikou committed:
Merge branch 'shipmap_chesh' of https://github.com/Cheshify/tgstation into shipmap_chesh

Fikou committed:
oops

Cheshify committed:
Fixed Linters?

Fikou committed:
fixes

Cheshify committed:
Merge branch 'shipmap_chesh' of https://github.com/Cheshify/tgstation into shipmap_chesh

Cheshify committed:
Linters: Done

Fikou committed:
more fixes

Fikou committed:
oops sorry chesh

Cheshify committed:
Lil Fixes

Cheshify committed:
Names the Asyum Moth Plush

Cheshify committed:
Renamed the Asylum Moth

Fikou committed:
last fixes

Fikou committed:
Merge branch 'shipmap_chesh' of https://github.com/Cheshify/tgstation into shipmap_chesh

Fikou committed:
oh vell

Fikou committed:
this may be worse but its future proof instead of relying on placement

Cheshify committed:
More Shuttles Part 1

Cheshify committed:
Merge branch 'shipmap_chesh' of https://github.com/Cheshify/tgstation into shipmap_chesh

Cheshify committed:
Last Shuttles + Json Updates

Cheshify committed:
forgot some turf_passthrough

Fikou committed:
trait fixes

Cheshify committed:
Misc Changes (Not Dead Yet!)

Fikou committed:
norandom

Fikou committed:
Merge branch 'shipmap_chesh' of https://github.com/Cheshify/tgstation into shipmap_chesh

Fikou committed:
fixes compile error

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
Keeping her up-to-date!

Cheshify committed:
Merge branch 'shipmap_chesh' of https://github.com/Cheshify/tgstation into shipmap_chesh

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
Landmarks fix?

Cheshify committed:
Saving Progress

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
We Back Boyos

Mothblocks commented on code/datums/shuttles.dm:
```diff
@@ -591,6 +596,13 @@
 	credit_cost = CARGO_CRATE_VALUE * 15
 
+/datum/map_template/shuttle/emergency/northstar
+	suffix ="northstar"
```
```suggestion
	suffix = "northstar"
```

Fikou commented on code/game/area/areas/station.dm:
```diff
@@ -333,6 +333,10 @@
 
+/area/station/hallway/secondary/exit/escape_pod
+	name = "\improper Escape Pod Bay"
+	icon_state ="escape_pods"
```
```suggestion
	icon_state = "escape_pods"
```

Cheshify commented:
There's only so much I can do with the ship right now. For the moment, there's no overmap or exploration, but once a number of optimization issues with the map are made, I'm comfortable with trying to expand what it can do ship-wise.

Right now it's a ship in style and in construction, but not mechanically.

Cheshify committed:
Update code/controllers/subsystem/processing/station.dm

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>

Cheshify committed:
Update code/game/area/areas/station.dm

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>

Cheshify committed:
newscaster fixes more to come

Cheshify committed:
Update code/datums/shuttles.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

Cheshify committed:
Merge branch 'shipmap_chesh' of https://github.com/Cheshify/tgstation into shipmap_chesh

Cheshify committed:
Should only fix request consoles.

Cheshify committed:
The rest of the bad vars

Cheshify committed:
Varedits are weird.

Cheshify committed:
Double lights

Cheshify commented:
I guarantee that there are some mild map issues, but it passes CI.

Cheshify commented:
Don't want this test-merged without me or before I've done local tests, please.

RegJackson commented:
![northstar](https://user-images.githubusercontent.com/91390185/229015339-96509ebc-97a1-492d-9b2f-7b453cf90c89.gif)

Rex9001 commented:
Return of the king! I played this once whilst it was tmd and the sheer size and fun of this map remains unrivaled on TG

Cheshify committed:
A few more final tweaks. She runs!

Cheshify commented:
There's still some issues, probably, minor ones, but I'd say she's ready for a *TEST* merge. I'd still like to be notified and present if that's the case so I can handle all map error reports.

theta060 commented:
tracking someone as a heretic will be really shitty

Cheshify committed:
Further Minor Improvements

Cheshify committed:
fixes unintentional separate pipenets

Cheshify commented on code/datums/station_traits/innate_traits.dm:
```diff
@@ -0,0 +1,107 @@
+		crewmember.shoes = null
+		qdel(old_shoes)
+		crewmember.equip_to_slot_if_possible(laceups, ITEM_SLOT_FEET, disable_warning = TRUE, initial = TRUE)
+	var/old_uniform = crewmember.w_uniform
+	if(old_uniform)
+		crewmember.w_uniform = null //to prevent side effects like dropping items, this is a temporary removal
+		qdel(old_uniform)
+		crewmember.equip_to_slot_if_possible(uniform, ITEM_SLOT_ICLOTHING, disable_warning = TRUE, initial = TRUE)
```
@Fikou You might have a better insight on this. From what I read, it might delete them depending on how and when accessories are added.

Cheshify committed:
cargoshuttle fixes evacshuttle design stationatmos

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
Updates the Emergency Shuttle

Cheshify commented:
Man, I almost forgot how hard it is to keep a testmerged map up to date.

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
various updates to med and botany

Cheshify committed:
posters and stuff too now that I think about it

Cheshify committed:
more meddling and improvements

Cheshify committed:
maint improvements + chapel runtime fix

Cheshify commented:
Had an issue with sparring sect chaplains, apparently since the chapel used the ``/monastery`` subtype it runtimed.

Cheshify committed:
mining shuttles courtesy of pumpkin0

Cheshify commented:
Aside from tests and polish, I don't know what else the map really needs.

Cheshify committed:
will this fix linters?

Rex9001 commented:
Small feedback: Cargo employees get the same outfit assistants get roundstart, a greenish star treck suit. Idk if this is intended

Cheshify commented:
> Small feedback: Cargo employees get the same outfit assistants get roundstart, a greenish star treck suit. Idk if this is intended

Assistants start in grey, cargo + service starts in green. Either there's an issue with the code, or you may be slightly colorblind.

Cheshify commented:
Just checked, cargo starts wearing green!

Cheshify committed:
trek uniform armor values and removes one light

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
update paths

Cheshify commented:
I don't even know why it had the tools tag

Cheshify committed:
adds bots and nav beacons

Cheshify commented:
> Small feedback: Cargo employees get the same outfit assistants get roundstart, a greenish star treck suit. Idk if this is intended

![image](https://user-images.githubusercontent.com/73589390/229676650-5debbe5c-d36b-47ac-a7d8-1502f1d1c389.png)

On testing, I'm afraid I don't know what you're talking about. Cargo wears green, assistants wear grey.

san7890 commented on code/datums/station_traits/innate_traits.dm:
```diff
@@ -0,0 +1,107 @@
+#define TREK_VARIATION_VOY "voy"
+#define TREK_VARIATION_ENT "ent"
+
+/datum/station_trait/trek
+	name = "Exploratory Uniforms"
+	trait_type = STATION_TRAIT_NORANDOM
```
@MrStonedOne FYI - This trait is automatically applied to this map by default via the JSON, which does bypass the `#GREY_ASSISTANTS` configuration. I don't think it matters too much at the end of the day, but given that you both introduced it and are the final say on configurationy stuff, do you think it matters?

Cheshify committed:
Various Small Updates

Cheshify commented:
![image](https://user-images.githubusercontent.com/73589390/229881886-1d1304e3-0899-4aed-9e81-079758268581.png)
Removed that line, I want to force players to interact with the coolant system

theta060 commented:
what are you even talking about coolent system i remember you saying stuff about them in the TMs just dump waste into space or a lone tile

Cheshify commented:
> what are you even talking about coolent system i remember you saying stuff about them in the TMs just dump waste into space or a lone tile

If you want to waste the SM gas, you can set it up in-round yourself. I don't want to encourage it by having it be a system, I'd rather give atmos techs actual job content and gas to use in said job content.

Cheshify commented:
Minor tangent but that comment really sucks. With that mindset, why do anything? Why not just have every map just be the same clone of each other where you're never tasked with anything, why have randomness even be an element, why have anything aside from the most basic boring and simple design where you're never challenged to try something new.

Rex9001 commented:
> > Small feedback: Cargo employees get the same outfit assistants get roundstart, a greenish star treck suit. Idk if this is intended
> 
> ![image](https://user-images.githubusercontent.com/73589390/229676650-5debbe5c-d36b-47ac-a7d8-1502f1d1c389.png)
> 
> On testing, I'm afraid I don't know what you're talking about. Cargo wears green, assistants wear grey.

Huh might just have been someone stealing, sorry for the trouble

Cheshify commented:
> Huh might just have been someone stealing, sorry for the trouble

Don't apologize! It got me to finally add armor values to the trek uniforms.

optimumtact commented:
I don't want more alternate uniforms and I don't see how to resolve this issue

thgvr commented:
The uniforms are already in the game

Cheshify commented:
> The uniforms are already in the game

There are eight uniforms (all made via gags) added in with the PR. The assistant grey and the service/cargo greens.

Rex9001 commented:
I dont really understand, is it just the outfits that make this massive map DNM or is it the map itself? Either case its kinda dumb, this map vastly outshines a lot of the in-rotation maps.

LemonInTheDark commented:
he's got an issue (debatable or not) with the outfits. doesn't apply to the map itself, and doesn't mean the map is dead.
It exists to prevent like eob from going "oh cool job's done" and just sending it before that issue gets resolved.

Cheshify commented:
Just waiting on a conversation with oranges. I pinged him in the player's project thread on the tgcord. I'm sure this'll be quick to resolve.

optimumtact commented:
I'm not sure it will be easy to resolve, I have concerns about setting a precedent that a different map gives wholesale approval to change the basic visual look of every job on the station with a different uniform, and I worry about where that will lead to, if we say yes to this one here now, just because the rest of the map is excellent, we are unable to say no later to others.

There are two aspects to this concern

First, maintainability - this adds a new variable to the massive ball of complexity that is debugging issues, a pretty mild one here since it's just uniforms, but one can easily imagine a uniform specific bug occurring only on this map

more importantly it's just asking for spriters to feel like they have to resprite two entire sets of uniforms anytime they want to make a visual change to a specific dept, because now there are two sets of specific uniforms in use.

The second concern is for new players, it's hard enough for htem to have to deal with an entirely new map layout from round to round, I think loading on top of that the fact that they also have to learn an entirely new set of uniforms and visual designs is asking too much and will lead to too much confusion that might turn them away from an already notoriously hard to pick up game.

Ultimately my view is that mappers have a lot of creative leeway with many things, but I do not believe that they get carte blanche to change the visual look of something as basic as the uniform of jobs just because it's a different map.

Unfortunately I do not think I am willing to compromise on this point, I think it's important to set a precedent here going forward, so it will be up to @ninjanomnom and @Mothblocks to decide if they want to overrule me on this

TheSmallBlue commented:
If I may give my 2 cents, I don't think it's that big of a deal.

The reason why North Star has differing uniforms is not "just because cesh could", it's because it's different from every other map we've had so far, North Star isn't a station, it's a ship, an entirely different setting from the space station one. If out intention is to have a Star Trek like retro space visual aesthetic, it's sensible and expected for ship and station uniforms to look and feel completely different. I heavily doubt we'll have more ship maps, so I heavily doubt this will be an issue that'll repeat itself, but even if we did it'd make sense if said ship map used the same trek uniforms for consistency and lore reasons. 

The uniforms themselves are just clothes, they function exactly like the default jumpsuits, the only way they could be broken are if all jumpsuits are broken, which has nothing to do with now north start itself implements them. The only way a north star specific uniform bug can come up is when the map is setting up and it applies the trek uniform trait or whatever it uses, in which case that would only affect north star. 

The uniforms themselves are clearly based on the star trek uniforms, and they're supposed to keep that aesthetic. If an artist wants to update the station science jumpsuit's aesthetic, they have no reason to do the same for the trek suits, since they're not supposed to be unique designs at all, they're supposed to stay a clear callback to star trek.

Saying that players will be turned away from the game just because they have a different piece of clothing is absurd. Again, the trek jumpsuits work exactly the same as the normal ones. You understand one, you understand the other. No player will leave because "I don't get why this spefic corner of my screen looks different than usual". By the same logic, make every job wear the same jumpsuit! Otherwise new players that are trying the janitor job will be very confused by the fact that the previously gray jumpsuit is now purple!

In my opinion, the pros of having an unique aesthetic to make the north star stand out even more than it already does (which in of itself brings in a breath of fresh air to what many consider the dull normal space station environment) heavily outweigh the con of "it might be hard to debug"

Cheshify commented:
> First, maintainability - this adds a new variable to the massive ball of complexity that is debugging issues, a pretty mild one here since it's just uniforms, but one can easily imagine a uniform specific bug occurring only on this map

> more importantly it's just asking for spriters to feel like they have to resprite two entire sets of uniforms anytime they want to make a visual change to a specific dept, because now there are two sets of specific uniforms in use.

This is a point I'm willing to somewhat agree on. It is a new set of uniforms, but you yourself say it's mild. I personally believe that for the sake of aesthetic design, the possible future potential of a mild bug is warranted? After all, we have plenty of feature PRs that can possibly add mild bugs, I'm not sure what makes this one any different. However, more workload for spriters is a concern that I can agree with. Spriters do a LOT of work in keeping the game looking good, and giving them more to do is worrying. I'll take this into account for possible compromises (I'm sure there are possible compromises here)

> The second concern is for new players, it's hard enough for htem to have to deal with an entirely new map layout from round to round, I think loading on top of that the fact that they also have to learn an entirely new set of uniforms and visual designs is asking too much and will lead to too much confusion that might turn them away from an already notoriously hard to pick up game.

This, SmallBlue put well. It's absurd to think that staying power will be an issue when crew uniforms are very mildly changed to reflect the setting. Nobody is going to look at a different uniform and outright leave and never come back, especially when vendors supply the job-standard uniforms.

> Ultimately my view is that mappers have a lot of creative leeway with many things, but I do not believe that they get carte blanche to change the visual look of something as basic as the uniform of jobs just because it's a different map.

I can respect this, though I think it's important to discuss WHY I chose this design. Apologies for sounding a bit like a pretentious artist type here, though this is the mindset I am in when working on this map,

When blocking out, designing, and eventually reworking parts of the map, my goal has always been to provide the distinct feeling of being "within an exploratory vessel". The uniforms are a minor but not insignificant part of that. They give players a tie that anchors themselves to the setting, using Star Trek as a 'subtext' for putting players in the mindset that they're in a different kind of environment. I think they add to the overall flavour of the map, and are also just fun to wear.

My intention has never been able to get away with "carte blanche" changes, but to provide players with that authentic experience. You'll see through the history of this project that I've struggled with "Metastationisms", where I catch myself (and others) suggesting that maybe things should be a little more standard, and I've fought to distance myself from it, in order to give players what I think is the best shipmap experience. I like the normal jumpsuits, they're iconic, well-sprited, and overall fun. I just think that when I'm going to make an authentic ship, we need something a little more exploratory.

----------Line Break This Is A Lot Of Writing----------

Okay, now. I know you said compromises weren't possible, but that's not something I can readily agree with. Here's a few thoughts and possible compromises.

- Remove the randomly chosen aspect of the uniforms, have one set of colored trek-inspired uniforms in order to make spriter's jobs significantly easier

- Remove the service and assistant uniforms, these are the only ADDED uniforms, the rest already exist within the game, I was told this could be a "too many sprites" issue, but that doesn't seem to be a part of your post.

- Finally, if all else fails. I'd be willing to rework or find some visual alternative to the uniforms

To note, I would VERY MUCH like to keep the innate station trait code. Having a system for mappers and coders to help differentiate maps beyond just having different layouts is a significant boon.

Mothblocks commented:
after some discussion i disagree with oranges on this point so we're throwing it to @ninjanomnom

Iamgoofball commented:
headcoder votes be like
![Multiple](https://user-images.githubusercontent.com/4081722/230260333-1caed5b5-6d77-4b8f-a820-dec8e1a94ec2.png)

ninjanomnom commented:
I've been thinking about this since yesterday when some of the technical concerns were being discussed in coderbus regarding the rsc costs. That's basically a non issue and were that the only consideration I would be fine with alternate uniforms per map. What oranges brings up though I find myself agreeing with, the maps we have in the repo should all follow similar designs graphically for consistency and maintenance sake.

We should not have alternate uniforms or any other such automatic resprites depending on the selected station map.

Cheshify commented:
Alright, and for keeping the innate station trait code?

Cheshify commented:
I have been informed that I cannot keep the framework.

Fikou commented:
Very disappointing decision.

optimumtact commented:
> I have been informed that I cannot keep the framework.

Who told you this? and can they please put their reasoning here.

Cheshify commented:
> > I have been informed that I cannot keep the framework.
> 
> Who told you this? and can they please put their reasoning here.

Mothblocks told me that since it would be unused here, it would need to be a separate PR to add innate station trait framework.

CMDR-Gungnir commented:
> What oranges brings up though I find myself agreeing with, the maps we have in the repo should all follow similar designs graphically for consistency and maintenance sake.
> 
> We should not have alternate uniforms or any other such automatic resprites depending on the selected station map.

But the thing is though, this creates the framework FOR that. This is a Ship, rather than a Station, so it has different shift-start uniforms. Then, if another Ship Map is made, they can share the expeditionary uniforms. We could have that consistency. But at the moment, this is our only Ship Map, and that's why it stands out on that front.

optimumtact commented:
please complain about the choice in a forum thread lets try to keep this to actionable things.

spockye reviewed (commented):
![image](https://user-images.githubusercontent.com/79304582/230425415-ef2460b1-e590-442e-8a4d-9c068c7ca98e.png)

pipe fuckery (tcomms storage) 

![image](https://user-images.githubusercontent.com/79304582/230425706-2cbfd8aa-ebd0-456e-b3cd-867bf99d52c0.png)

missing pipes (HFR atmos)

![image](https://user-images.githubusercontent.com/79304582/230387789-69284183-74d3-4231-8183-57b7dcc74fcc.png)

missing pipe (departures)

![image](https://user-images.githubusercontent.com/79304582/230388494-5e1107bb-587a-45a9-ae27-19f145efb0d0.png)

missing pipes (3rd floor central starboard maints)

![image](https://user-images.githubusercontent.com/79304582/230388933-97cb7529-1e14-4687-bc4f-c32e59398c75.png)

missing wires (armoury)

![image](https://user-images.githubusercontent.com/79304582/230389248-88642dbf-a9ce-4e72-8640-86dec30eb598.png)

missing wire (entertainment centre)

![image](https://user-images.githubusercontent.com/79304582/230416801-c4371410-340a-4e91-b37b-2b61cc3cacc1.png)

missing wire (cargo bay)

![image](https://user-images.githubusercontent.com/79304582/230416967-4c7b63f3-5aa4-49fc-9359-66522be3ef03.png)

wacky wires (cargo bay)

![image](https://user-images.githubusercontent.com/79304582/230418008-d2e16336-45b2-400c-8989-7fc3e9464cc2.png)

missing zone (might just be on my side)

![image](https://user-images.githubusercontent.com/79304582/230422250-fafc6a1e-5b2a-4635-a717-43886db7f4a7.png)

zone fuckery

![image](https://user-images.githubusercontent.com/79304582/230418864-ed25dd72-aa44-49c5-a721-9d7d441e2973.png)

disposals fuckery (second floor fore hallway)

![image](https://user-images.githubusercontent.com/79304582/230419229-c80f36f5-e19a-4900-8ce6-642e54eba936.png)

disposals fuckery (chapel)

![image](https://user-images.githubusercontent.com/79304582/230419369-4cd8f9c3-62ae-4ad1-939f-9f868e07e59d.png)

missing disposals (ordenance testing lab)

![image](https://user-images.githubusercontent.com/79304582/230419930-26968aa6-d1a9-46c1-8fa3-2a181205dc68.png)

shouldn't there be nearstation on this?


the map looks rad. Amazing work man!

Cheshify commented:
> the map looks rad. Amazing work man!

I greatly appreciate the review, I'm putting out a fix for all of these right now

Cheshify commented:
OH, a few of those wire issues are because I use the shocked grille spawner! So they're not really broken.
![image](https://user-images.githubusercontent.com/73589390/230457710-e84b8a7c-bfd4-4ef7-affa-0d8bafbd52dd.png)

Cheshify committed:
spockye review + armory fun

Cheshify commented:
Added some "emergent gameplay opportunities" to the armory.

Cheshify committed:
Removes Innate Traits

13spacemen commented:
Please fix the shutters on this map, all of them are facing south even though shutters have directions
![shutters1](https://user-images.githubusercontent.com/46101244/230475005-f4ed2bac-99af-4f3b-9baa-d00975565d8a.png)
![shutters2](https://user-images.githubusercontent.com/46101244/230475016-1e8a0f9b-377e-47f9-8ab0-b2e90c12a549.png)


For an example of correct shutter dirs you can see Metastation

The shutters here should be facing North and West but they're all South

Cheshify commented:
We have directional shutters now? Sure.

Cheshify committed:
A few directional shutters (but not too many)

Cheshify commented:
There are a few shutters I didn't rotate just because I didn't think they looked good, but I did a few you pointed out

Cheshify committed:
Also reverts positive_traits.dm

Mothblocks commented on code/modules/clothing/under/trek.dm:
```diff
@@ -1,3 +1,4 @@
+
```
```suggestion
```

Mothblocks commented:
Removing DNM since alternate outfits are removed

spockye reviewed (commented):
![image](https://user-images.githubusercontent.com/79304582/230503455-1118c00a-045a-4e09-9018-90a41b0f8872.png)

im pretty sure this should be the public version of the console (it's also locked behind an access door)

![image](https://user-images.githubusercontent.com/79304582/230505088-27cae7a0-5eec-41e9-a50a-e4f86d4ce8d0.png)

(hot tip hot tip!!!) a pretty nice way of creating a effect of railings being above something, is to set it's layer to 3.1 (4.1 to also be above plants)

Cheshify committed:
mines + random mine spawner

Cheshify committed:
spockye reviews

Cheshify commented:
> im pretty sure this should be the public version of the console (it's also locked behind an access door)
> 
> (hot tip hot tip!!!) a pretty nice way of creating a effect of railings being above something, is to set it's layer to 3.1 (4.1 to also be above plants)

Another fantastic review, all changes made!

Cheshify committed:
more dirt+dust - SM no longer hotwired roundstart

Cheshify commented:
Whoops! Minor issue with wiring I caught last-minute. I was always wondering why the SMES charged slow and why the shocked grilles would instakill sometimes...

Well, it's been fixed luckily

Cheshify committed:
Update code/modules/clothing/under/trek.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
Merge branch 'shipmap_chesh' of https://github.com/Cheshify/tgstation into shipmap_chesh

Cheshify committed:
Second time fixing landmarks...

Addust commented:
> Please fix the shutters on this map, all of them are facing south even though shutters have directions ![shutters1](https://user-images.githubusercontent.com/46101244/230475005-f4ed2bac-99af-4f3b-9baa-d00975565d8a.png) ![shutters2](https://user-images.githubusercontent.com/46101244/230475016-1e8a0f9b-377e-47f9-8ab0-b2e90c12a549.png)
> 
> For an example of correct shutter dirs you can see Metastation
> 
> The shutters here should be facing North and West but they're all South

irrelevent but Funni Terry In-Round Pictures of Me Having A Fucking Skill Issue

Cheshify committed:
changes the 644 to 755

Cheshify commented:
Alright, got rid of those files I didn't touch but were changed anyhow.

Rex9001 commented:
Cargo techs can access the grav gen for some reason.... but this might not be map specific and just a general access bug

Cheshify committed:
gravgen access

Cheshify commented:
> Cargo techs can access the grav gen for some reason.... but this might not be map specific and just a general access bug

fixed

Maurukas reviewed (changes requested):
Alright. This has already been polished quite a bit, and that shows. There's a lot that's different here and a lot that I like. I managed to find a few things, largely cosmetic as I went through.

I'll poke eob about things.

The SM is bolted at roundstart. While engineers can play with wires, a button to unbolt it nearby seems to me like it would be a good addition.

<details>
<summary>There's a fax machine on the floor of engineering next to second fax machine</summary>

![image](https://user-images.githubusercontent.com/66576896/230696442-9ffeea88-3ce3-4ba5-aec2-f2ac03443e60.png)
</details>

<details>
<summary>Some pressure sensors in the SM waste handling would be appreciated</summary>

This is non-standard and increased visibility into what's happening in the pipes would be helpful.

![image](https://user-images.githubusercontent.com/66576896/230696567-ffc29172-672f-4d7a-8904-bf4ef5693ac6.png)
</details>

<details>
<summary>These air supply pipes should be visible</summary>

![image](https://user-images.githubusercontent.com/66576896/230697126-04b4cf77-c81c-425a-a934-0e167024357c.png)
</details>
<details>
<summary>The.. uh. Tinting didn't work</summary>

![image](https://user-images.githubusercontent.com/66576896/230697241-594b538c-1c3e-48a9-8cbf-f0dc06d53ccf.png)
</details>

Cheshify committed:
requested changes made

Cheshify commented:
Requested changes made:
I also removed the extra fax and made those two pipes visible.

<details>
<summary>Theater bathroom made grimier and now with a door bolt. Tossed some more posters around too.</summary>

Red arrows point from actual position to viewed position.
![image](https://user-images.githubusercontent.com/73589390/230699769-8a0ada45-cbdf-4564-a626-2a619e75de64.png)

</details>

<details>
<summary>SM chamber has a door bolt.</summary>

![image](https://user-images.githubusercontent.com/73589390/230699862-f96e9204-fc3b-45be-8d3d-7c3f72f7f170.png)

</details>

<details>
<summary> SM waste now has some meters so you can easily check up on coolant temperature and siphoned gas</summary>

I also added some comms-building stuff in the little telecomms storage nearby, just to fill out an empty rack
![image](https://user-images.githubusercontent.com/73589390/230699941-3b41bbef-7ba8-4778-9ce1-426c7aeef2f5.png)

![image](https://user-images.githubusercontent.com/73589390/230699881-30761493-fdcd-4339-a7e7-36ef3d6941ac.png)

</details>

When this is merged, could I make a secondary PR for a map plaque detailing date of merge?

Time-Green commented:
There are only 2 deepspace z-levels, which are completely oversaturated with ruins (I couldnt find a clean bit of space that didnt have ruins)
![image](https://user-images.githubusercontent.com/7501474/230739489-4017ea0d-7834-49a6-b3a4-ea254858b24f.png)

optimumtact commented:
probably need to be able to set a per map ruin budget to avoid this issue since most other maps have way more deep space z's

Cheshify commented:
> There are only 2 deepspace z-levels, which are completely oversaturated with ruins (I couldnt find a clean bit of space that didnt have ruins)
> 
> picture

I wouldn't have issues with just removing the extra space levels honestly. The map is plenty large as is and keeping the action focused to the station + lavaland may be smart?

Cheshify commented:
I'll wait on maptainer input

CMDR-Gungnir commented:
The Bartender's mini-kitchen's fridges aren't unlockable by said Bartender. Also, while I get that it's meant to be a MINI kitchen, it lacks an oven, which would make a pastry-cafe extremely difficult.

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify commented:
> The Bartender's mini-kitchen's fridges aren't unlockable by said Bartender. Also, while I get that it's meant to be a MINI kitchen, it lacks an oven, which would make a pastry-cafe extremely difficult.

Fixed the access on the fridges. An oven is a bit too much for the minikitchen. I'd suggest working with chefs to see that done.

Cheshify committed:
More QOL

MrStonedOne commented:
> probably need to be able to set a per map ruin budget to avoid this issue since most other maps have way more deep space z's

just a modifier. make it say 0.3-0.4 for north star.

DATA-xPUNGED commented:
There is a door that goes from the main cargo to the ORM that miners can't acces, but they CAN access the door that goes from the decontamination chamber to the ORM, so i have to go around it every time.

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
Update Paths

Cheshify committed:
more QOL

MrrFiish commented:
Over by the arrivals shuttle on the left side and then a bit down there's a floating lightbulb in maints, literally unplayable wtf.

Cheshify committed:
CI Fixes + Signage

Cheshify committed:
better barriers and QOL

Cheshify commented:
Got some complaints about the cargo barriers so there's a new subtype to buff em up.

13spacemen commented:
The green bridge pipes in atmos are rotated vertically instead of horizontally and dont connect
Also the window shutters in Incinerator should be rotated to face west

Cheshify commented:
> Had an issue with space ruins not spawning at all on the TM. Zero distant signals on GPS and, for as much of the deep space levels I could sweep, they were completely empty.

Intentional, I don't really want offstation content being the focus when the ship is so large, and some kind of bug was causing ruins to spawn jam-packed next to eachother. Disabled for now, maybe I'll bring back ruins later.

Cheshify commented:
> The green bridge pipes in atmos are rotated vertically instead of horizontally and dont connect Also the window shutters in Incinerator should be rotated to face west

![image](https://user-images.githubusercontent.com/73589390/231261567-f43c42c0-0822-4015-8e90-f341dc516960.png)
I don't know what you mean about the bridge pipes for atmos, they're all hoz. to me?

The shutters look a bit off when rotated like that, I'll try it out for now but I'm not exactly sure if I'll keep it.

Cheshify commented:
> > Intentional, I don't really want offstation content being the focus when the ship is so large, and some kind of bug was causing ruins to spawn jam-packed next to eachother. Disabled for now, maybe I'll bring back ruins later.
> 
> Maybe in a future PR space travel could be added as the only mean to reach ruins

That sounds awesome

Cheshify commented:
> > > Had an issue with space ruins not spawning at all on the TM. Zero distant signals on GPS and, for as much of the deep space levels I could sweep, they were completely empty.
> > 
> > 
> > Intentional, I don't really want offstation content being the focus when the ship is so large, and some kind of bug was causing ruins to spawn jam-packed next to eachother. Disabled for now, maybe I'll bring back ruins later.
> 
> Didn't one of the reviewers say something along the lines of that ruin spawns should be reduced, not removed?
> 
> For one, certain research isn't possible unless certain things are found. But even if you throw a few alien/experimentals in maintenance, just cutting out a gameplay loop like that is pretty callous. A bit like if you decided, say, a virology area isn't needed because who plays that and with a pandemic machine in chemistry means you're covered for cures.

Space Loot isn't a mandatory piece of content. It's fun optional side content. The map is 4 levels tall and has a very fun maint to explore. Lavaland still exists for ruin delving.

YehnBeep commented:
Very little content is mandatory ultimately. I just had higher hopes that you wouldn't cut stuff because people aren't playing the way you want them too. But oh well, I suppose.

Cheshify committed:
every gets a sink!

Cheshify committed:
Ordnance better

Cheshify committed:
misc fixes and changes and whatnot

Cheshify commented:
If someone else complains about something being non-standard or different than meta, or complains about maint being too dangerous I'm adding more land mines to maint.

TribeOfBeavers commented:
Good map! ^-^
Very unique, I've enjoyed all the rounds I've played on it so far (And I've only played like 2 departments in that time) 

My one suggested change would be having the doors leading into engineering only be linked to the ones further in from them instead of all of them, if that makes sense. As it is, when people are trying to use the parallel sets of doors they end up closing them on each other a bunch and it gets annoying getting groups of people in or out. 

For example:

A--B
-----
C--D

Have A and B linked, and C and D linked. As is, opening A and then C will cause A to close early. But it would maybe be a little more streamlined if A only closed B (Or B and D I suppose, if that's possible). Or is there a specific reason they're set up the way they are?

Cheshify committed:
removes single access helper

HalfAwake691 commented:
I dislike how disconnected chem is from the rest of medical, Sometimes people dont even comeby because of how off to the side it is. Plus the walk from pharma/Chem is waaaaay to long just to deliver chems. I wish that reception desk was pharma instead sometimes so ican at least help people/open doors for them if needed and people can order stuff and what not.

Cheshify commented:
> after getting used to it, I can definitely say that it's goddamned awesome and the best map I've seen in literal years.
> 
> re the security outposts though, they don't have prisoner lockers. they really don't.
> 
> ![4wXmLpsEn2](https://user-images.githubusercontent.com/39802339/231325787-babd7de7-7469-4b0c-bb94-48f215f1d622.png)

I just added them.

Cheshify commented:
> I dislike how disconnected chem is from the rest of medical, Sometimes people dont even comeby because of how off to the side it is. Plus the walk from pharma/Chem is waaaaay too long just to deliver chems. I wish that reception desk was pharma instead sometimes so i can at least help people/open doors for them if needed and people can order stuff and what not.
> 
> * Emerald Bromson

Pharmacy is like a 20 tile walk to the treatment room, and it has an access desk to the hall. I don't know what you mean by "too far away".

Cheshify commented:
Also I severely nerfed how many grilles are in maint, so you may be playing on an old commit.

HalfAwake691 commented:
> > I dislike how disconnected chem is from the rest of medical, Sometimes people dont even comeby because of how off to the side it is. Plus the walk from pharma/Chem is waaaaay too long just to deliver chems. I wish that reception desk was pharma instead sometimes so i can at least help people/open doors for them if needed and people can order stuff and what not.
> > 
> > * Emerald Bromson
> 
> Pharmacy is like a 20 tile walk to the treatment room, and it has an access desk to the hall. I don't know what you mean by "too far away".

It's like the only pharma thats not connected to the front lobby its just kinda pushed off to the side, I would like to see the front desk area made into pharma at least personally. Add a door that leads into the front like kilo and it would be 10/10 imho.

![image](https://user-images.githubusercontent.com/121485469/231333870-5c461be7-40c3-4450-8fb5-e2c5e2dacccb.png)

Cheshify committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into shipmap_chesh

Cheshify committed:
up to date and oproom shutters

optimumtact commented:
Say good bye map tick

TheVekter commented:
WE'RE SO BACK

DAKKA-WAAAGH commented:
IT'S NOT OVER

Cheshify commented:
We're so fucking back boys

Zonespace27 commented:
HOLY SHIT ITS HERE WOOOOOOOOOOOOO

Pumpkinoe commented:
WE ARE SO FORWARD
